code/datums/datum.dm

```diff
@@ -9,6 +9,15 @@
 	/// Datum level flags
 	var/datum_flags = NONE
 
+	/**
+	  * Any datum registered to receive observations from this datum is in this list
+	  *
+	  * Lazy associated list in the structure of `observation:registree/list of registrees`
+	  */
+	var/list/observations
+	/// Lazy associated list in the structure of `observations:proctype` that are run when the datum receives that observation
+	var/list/list/datum/callback/observation_procs
```

Comment:
Strictly typing this list may lead to wackiness, I’m not 100% sure on how well BYOND actually deals with these things. I believe the convention is to use a loosely typed assoc list with type checking as needed.

Reply:
Again, if not already existent, a wrapper proc to add an event listener could be nice here.

Reply:
Byond have dynamic typization.
Example create var/thing = list()
thing will contain list, but don't have memory access for work with list (thing.len, thing.Copy, thing.Cut and etc)